samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -279,8 +285,17 @@ protected Config getConfig() {
     return config;
   }
 
+  private void addDropProducerErrorsConfig() {
+    Map<String, String> newConfig = new HashMap<>();
+    if (config.get(DROP_PRODUCER_ERRORS) == null) {
+      newConfig.put(DROP_PRODUCER_ERRORS, "true");
+      config = new MapConfig(ImmutableList.of(config, newConfig));
```

Comment:
Cleaner to use `new MapConfig(config, ImmutableMap.of("drop.xyz.errors", "baz"))` here? A nice thing is that you don;t need this boiler-plate write to the single-element hashmap.

Reply:
Got it. Thank you :)